### PR TITLE
login by google user

### PR DESCRIPTION
**Pull request by koropalov**

### CODE REVIEW

File: src/pages/login/login.js [hunk 1]
```diff
@@ -50,6 +54,22 @@ const Login = () => {
       .required(' ')
   });
 
+  useEffect(() => {
```
Comment (maksym-strus) [inline src/pages/login/login.js:57]: If we need it to init only once you should write it with empty dependencies array: **useEffect(()=>{...},**[]**)**

File: src/redux/user/user.actions.js [hunk 2]
```diff
@@ -35,6 +36,11 @@ const loginUser = (payload) => ({
   payload
 });
 
+const loginByGoogle = (payload) => ({
+  type: GOOGLE_USER,
```
Comment (maksym-strus) [inline src/redux/user/user.actions.js:40]: You need to use a more understandable type name for this action (like **LOGIN_USER_BY_GOOGLE** or **LOGIN_BY_GOOGLE**)

File: src/redux/user/user.types.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ export const SET_USER = 'SET_USER';
 export const SET_USER_ERROR = 'SET_USER_ERROR';
 export const SET_USER_LOADING = 'SET_USER_LOADING';
 export const LOGIN_USER = 'LOGIN_USER';
+export const GOOGLE_USER = 'GOOGLE_USER';
```
Comment (maksym-strus) [inline src/redux/user/user.types.js:5]: You need to use a more understandable type name for this action (like LOGIN_USER_BY_GOOGLE or LOGIN_BY_GOOGLE)

File: src/redux/user/user.sagas.js [hunk 4]
```diff
@@ -395,4 +428,5 @@ export default function* userSaga() {
   yield takeEvery(UPDATE_USER, handleUpdateUser);
   yield takeEvery(SEND_CONFIRMATION_EMAIL, handleSendConfirmation);
   yield takeEvery(GET_USER_ORDERS, handleGetUserOrders);
+  yield takeEvery(GOOGLE_USER, handleGoogleUserLogin);
```
Comment (maksym-strus) [inline src/redux/user/user.sagas.js:431]: You need to use a more understandable type name for this action (like **LOGIN_USER_BY_GOOGLE** or **LOGIN_BY_GOOGLE**)

File: src/pages/login/login.js (hunk 1 above)
Comment (koropalov) [reply src/pages/login/login.js:57]: Fixed

File: src/redux/user/user.actions.js (hunk 2 above)
Comment (koropalov) [reply src/redux/user/user.actions.js]: Fixed

File: src/redux/user/user.types.js (hunk 3 above)
Comment (koropalov) [reply src/redux/user/user.types.js]: Fixed

File: src/redux/user/user.sagas.js (hunk 4 above)
Comment (koropalov) [reply src/redux/user/user.sagas.js]: Fixed

File: src/pages/login/login.js [hunk 5]
```diff
@@ -50,6 +54,22 @@ const Login = () => {
...
+    window.gapi.load('auth2', function () {
+      window.gapi.auth2.init({
+        client_id: process.env.REACT_APP_GOOGLE_CLIENT_ID
+      });
+    });
+  }, []);
+
+  const singIn = () => {
+    const GoogleAuth = window.gapi.auth2.getAuthInstance();
```
Comment (TorskyM) [inline src/pages/login/login.js:66]: Rename `GoogleAuth` to `googleAuth`

File: src/redux/user/user.sagas.js [hunk 6]
```diff
@@ -75,6 +76,38 @@ export const loginUser = (data) => {
...
+      lastName,
+      email,
+      credentials{
+        source,
+        tokenPass
+      }
+  token
+} 
+
+}
```
Comment (TorskyM) [inline src/redux/user/user.sagas.js:96]: Align this mutation to make it more readable

File: src/pages/login/login.js [hunk 7]
```diff
@@ -50,6 +54,22 @@ const Login = () => {
       .required(' ')
   });
 
+  useEffect(() => {
+    window.gapi.load('auth2', function () {
```
Comment (TorskyM) [inline src/pages/login/login.js:58]: Use the arrow function here

File: src/pages/login/login.js (hunk 5 above)
Comment (koropalov) [reply src/pages/login/login.js]: fixed

File: src/redux/user/user.sagas.js (hunk 6 above)
Comment (koropalov) [reply src/redux/user/user.sagas.js]: fixed

File: src/pages/login/login.js (hunk 7 above)
Comment (koropalov) [reply src/pages/login/login.js]: fixed

File: src/redux/user/user.sagas.js [hunk 8]
```diff
@@ -75,6 +76,38 @@ export const loginUser = (data) => {
...
+        id_token: payload.id_token
+      }
+    );
+    yield put(setUser(user.data.googleUser));
+    yield setToLocalStorage('accessToken', user.data.googleUser.token);
+    yield put(setUserLoading(false));
+    yield put(push('/profile'));
+  } catch (error) {
+    yield put(setUserError(error.message.replace('GraphQL error: ', '')));
+  }
```
Comment (ahd3r) [inline src/redux/user/user.sagas.js:108]: It is a good practice to have a `final` cell in redux, to stop `loading`, please add it.

Comment (koropalov) [reply src/redux/user/user.sagas.js:109]: Fixed

File: src/pages/login/login.js [hunk 9]
```diff
@@ -50,6 +54,22 @@ const Login = () => {
...
+      window.gapi.auth2.init({
+        client_id: process.env.REACT_APP_GOOGLE_CLIENT_ID
+      });
+    });
+  }, []);
+
+  const singIn = () => {
+    const googleAuth = window.gapi.auth2.getAuthInstance();
+    googleAuth.signIn().then((googleUser) => {
+      const id_token = googleUser.getAuthResponse().id_token;
```
Comment (TorskyM) [inline src/pages/login/login.js:68]: Don't use underscore in your `const` declaration

Comment (koropalov) [reply src/pages/login/login.js]: Fixed

**Review by TorskyM — APPROVED** [review]

**Review by maksym-strus — APPROVED** [review]

**Pull request merged**